examples/technical_blogs/tartan_blog/build_and_run_networks.py

```diff
@@ -0,0 +1,187 @@
+os.environ["CMAKE_C_COMPILER"] = os.path.join(args.oneapi_base_toolkit, "compiler", "linux", "bin-llvm", "clang")
+os.environ["LD_LIBRARY_PATH"] = f"{os.path.join(args.oneapi_base_toolkit, 'compiler', 'linux', 'lib')}:{os.path.join(args.oneapi_base_toolkit, 'compiler', 'linux', 'compiler', 'lib', 'intel64_lin')}:{os.environ.get('LD_LIBRARY_PATH', '')}"
+os.environ["OCL_ICD_FILENAMES"] = f"{os.path.join(args.oneapi_construction_kit_build_dir, 'lib', 'libCL.so')}"
+
+
+def run_cmake_and_ninja(cmake_command, build_dir):
+    try:
+        subprocess.run(cmake_command, check=True)
+        subprocess.run(["ninja", "-C", build_dir], check=True)
+    except subprocess.CalledProcessError as e:
+        print(f"Error executing command: {e}", file=sys.stderr)
+        sys.exit(1)
+
+# Build portBLAS
+def build_portBLAS():
+    script_directory = os.getcwd()
+    portBlas_build_dir = os.path.join(script_directory, "build_portBlas")
+    
+    cmake_portBlas_command = [
+        "cmake",
+        f"-B {portBlas_build_dir}",
+        f"{args.portBLAS_source}",
+        "-GNinja",
+        "-DSYCL_COMPILER=dpcpp"
+    ]
+
+    run_cmake_and_ninja(cmake_portBlas_command, portBlas_build_dir)
+
+# Build SYCL-DNN
+def build_sycl_dnn():
+    script_directory = os.getcwd()
+    sycl_dnn_build_dir = os.path.join(script_directory, "build_sycl_dnn")
+    
+    cmake_sycl_dnn_command = [
+        "cmake",
+        f"-B {sycl_dnn_build_dir}",
+        f"{args.SYCL_DNN_source}",
+        "-GNinja",
+        f"-DCMAKE_CXX_COMPILER={os.environ['CMAKE_CXX_COMPILER']}",
+        "-DSNN_BUILD_BENCHMARKS=OFF",
+        "-DSNN_BENCH_SYCLBLAS=OFF",
+        "-DSNN_BUILD_DOCUMENTATION=OFF",
+        f"-DSyclBLAS_DIR='{os.path.join(script_directory, 'build_portBlas')}'"
+    ]
+
+    run_cmake_and_ninja(cmake_sycl_dnn_command, sycl_dnn_build_dir)
+
+
+def download_and_convert_model(dir, file_url, destination_file, h5_to_bin_script):
+    #Check if the model dir already exists
+    path = os.path.join(os.getcwd(), dir)
+    if os.path.exists(path):
+        print(f"'{path}' already exists. Skipping creating '{path}'.")
+    else:
+        os.mkdir(path)
+        print(f"Directory '{path}' created successfully.")
+
+    # Check if the file already exists
+    if os.path.exists(destination_file):
+        print(f"'{destination_file}' already exists. Skipping download.")
+    else:
+        try:
+            response = requests.get(file_url)
+            response.raise_for_status()
+            with open(destination_file, "wb") as file:
+                file.write(response.content)
+            print(f"Downloaded '{destination_file}' successfully.")
+        
+            subprocess.run(["python", h5_to_bin_script, destination_file], check=True)
+            print(f"Converted '{destination_file}' to binary successfully.")
+            os.chdir("..")  # Move up one level to the parent directory
```

Comment:
could we avoid the chdir by just making the destination_file path include the sub directory?